### PR TITLE
Generate the NCN layer

### PR DESCRIPTION
**Pull request by Pete-Y-CS**

### CODE REVIEW

File: layers/generate_layers.py [hunk 1]
```diff
@@ -173,6 +178,34 @@ def fixProps(inputProps):
 
     convertGeoJsonToPmtiles(f"{tmp}/mrn.geojson", "output/mrn.pmtiles")
 
+def makeNationalCycleNetwork():
+    tmp = "tmp_ncn"
+    ensureEmptyTempDirectoryExists(tmp)
+
```
Comment (dabreegster) [inline layers/generate_layers.py:184]: For posterity, a comment linking to https://data-sustrans-uk.opendata.arcgis.com/ useful. The arcgis URL isn't discoverable or could change with another version

File: layers/generate_layers.py [hunk 2]
```diff
@@ -173,6 +178,34 @@ def fixProps(inputProps):
...
+    # Get the shapefile
+    run(
+        [
+            "wget",
+            "https://services5.arcgis.com/1ZHcUS1lwPTg4ms0/arcgis/rest/services/National_Cycle_Network_Public/FeatureServer/0/query?outFields=*&where=1%3D1&f=geojson",
```
Comment (dabreegster) [inline layers/generate_layers.py:189]: How'd you get this link? I got `https://opendata.arcgis.com/api/v3/datasets/5defd254e78745bfb12d0456abc1bcf1_0/downloads/data?format=geojson&spatialRefId=4326&where=1%3D1` by going to https://data-sustrans-uk.opendata.arcgis.com/datasets/5defd254e78745bfb12d0456abc1bcf1_0/explore, clicking download, going down to geojson, choosing the file that's already there

File: layers/generate_layers.py [hunk 3]
```diff
@@ -173,6 +178,34 @@ def fixProps(inputProps):
 
     convertGeoJsonToPmtiles(f"{tmp}/mrn.geojson", "output/mrn.pmtiles")
 
+def makeNationalCycleNetwork():
+    tmp = "tmp_ncn"
+    ensureEmptyTempDirectoryExists(tmp)
+
+    # Get the shapefile
```
Comment (dabreegster) [inline layers/generate_layers.py:185]: Remove comment, this isn't a shapefile

File: layers/generate_layers.py (hunk 2 above)
Comment (Pete-Y-CS) [reply layers/generate_layers.py:189]: Got the same link as you, will use it instead and rerun

File: layers/generate_layers.py [hunk 4]
```diff
@@ -173,6 +178,30 @@ def fixProps(inputProps):
...
+            "-O",
+            f"{tmp}/national_cycle_network.geojson",
+        ]
+    )
+
+
+    def fixProps(inputProps):
+        propsToRemove = ["LinkNo", "Lighting",  "RoadClass", "SHAPE_Length", "GlobalID", "SegmentID"]
+        for prop in propsToRemove:
+            inputProps.pop(prop)
```
Comment (dabreegster) [inline layers/generate_layers.py:199]: I've never seen `pop` on a dictionary, only ever to remove the last thing from a list. But it works! TIL more python

**Review by dabreegster — APPROVED** [review]
Everything looks good here! I re-ran locally and copied to S3. Looking at the frontend PR now

**Pull request merged**